src/app/components/frequency-picker/frequency-picker.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { MatChip } from '@angular/material/chips';
+
+enum Frequency {
+  daily = 'daily',
+  weekly = 'weekly',
+  monthly = 'monthly',
+}
+
+enum DayOfMonth {
+  first = 'First',
+  last = 'Last',
+  specificLast = 'SpecificLast',
+  custom = 'Custom',
+}
+
+@Component({
+  selector: 'app-frequency-picker',
+  templateUrl: './frequency-picker.component.html',
+  styleUrls: ['./frequency-picker.component.scss'],
+})
+export class FrequencyPickerComponent implements OnInit {
+  frequencyEnum = Frequency;
+  DayOfMonthEnum = DayOfMonth;
+  today: Date = new Date();
+
+  daysOfWeek: Map<number, string> = new Map([
```

Comment:
Can we move this code in util files so everyone can reuse this piece of code

Reply:
I'm not sure if "today" will work in utils since the lifetime of today will be tied to the whole app instead of our component. Or is it created for each component when untils is imported in that component?